docs/fsharp-cheatsheet.md

```diff
@@ -540,17 +538,156 @@ Another way of implementing interfaces is to use *object expressions*.
+
+    [<AutoOpen>]
+    module MyModule =
+        type DU = A | B | C
+
+    let du = A
+
+### Accessibility Modifiers
+
+F# supports `public`, `private` (restraining the element to its containing `type` or `module`) and `internal` (restraining the element to its containing assembly).
+It can be applied to `module`, `let`, `member`, `type`, and `new`.
+
+With the exception of `let` bindings in a class `type`, everything defaults to `public`.
+
+| Element                                                           | Example with Modifier                      |
+|-------------------------------------------------------------------|--------------------------------------------|
+| Module                                                            | `module internal MyModule =`               |
+| Module .. `let`                                                   | `let private value =`                      |
+| Record                                                            | `type internal MyRecord = { id: int }`     |
+| Record [ctor](#CodeOrganization_PrivateConstructors)              | `type MyRecord = private { id: int }`      |
+| Discriminated Union                                               | `type private MyDiscUni = A \| B`          |
+| Discriminated Union [ctor](#CodeOrganization_PrivateConstructors) | `type MyDiscUni = internal A \| B `        |
+| Class                                                             | `type internal MyClass() =`                |
+| Class [ctor](#CodeOrganization_PrivateConstructors)               | `type MyClass private () =`                |
+| Class Additional [ctor](#CodeOrganization_PrivateConstructors)    | `internal new() = MyClass("defaultValue")` |
+| Class .. `let`                                                    | *Always private. Cannot be modified*       |
+| `type` .. `member`                                                | `member private _.classMember =`           |
+
+<a name="CodeOrganization_PrivateConstructors"></a>
+##### Private Constructors
+
+Limiting `type` constructors (ctor) accessibility is a good way to enforce value integrity.
+
+Example of Single-case Discriminated Union with a `private` constructor:
+
+    type UnitQuantity =
+        private UnitQuantity of int
+        with
+        static member private MaxQty = 100
+        static member Create (qty:int) : Result<UnitQuantity, string> =
+            if qty <= UnitQuantity.MaxQty
+            then Ok (UnitQuantity qty)
+            else Error $"UnitQuantity cannot be more than {UnitQuantity.MaxQty}"
+    
+    let uQty = UnitQuantity.Create 50
+    match uQty with
+    | Ok (UnitQuantity qty) -> printfn $"Good: {qty}"
+    | Error errStr -> printfn $"Bad: {errStr}"
+
+Example of a class with a `private` constructor:
+
+    type MyClass private (count:int) =
+        member this.Count = count
+        static member CreateInstance (cnt: int) : MyClass option =
```

Comment:
if you had an example with a private primary and a public validating new, then this could be a sister module 
```
module MyClass =
    let tryCreate cnt =
        if cnt <= 0 then None
        else MyClass(cnt)
```
(there should be a section on types with companion modules being a thing vs lots of static members)        
also the `new` here will raise a warning in IDEs as they are only 'required' for things with IDisposable

Reply:
I just took this out. Seemed redundant and non-idiomatic to show the class side.